tests/unit/test_xtcedef.py

```diff
@@ -1796,7 +1796,7 @@ def test_boolean_parameter_type(xml_string, expectation):
             xtcedef.BinaryDataEncoding(fixed_size_in_bits=1)),
          {},
          xtcedef.PacketData(0b0011010000110010010100110000000001001011000000000100100100000000.to_bytes(length=64, byteorder='big')),
-         b'', False),
+         b'\x00', True),
```

Comment:
I think I messed up previously 🐑 ,
https://github.com/medley56/space_packet_parser/commit/15af144257dda66744bbcbc9595d0dd0107dc495#diff-56c0e979a3995704b94f150fb764e7e524b03a092ad029050c1d99c6c6115cd2R1799

I'm now going back on what I said the last time and think this should actually be as you had it before. We are saying this is `BinaryDataEncoding`, which means we want bytes back. The last update I made squeezed the data down into the smallest number of bytes. Whereas with this change we are padding with 0s and expanding to fit the data within a byte.

You are emitting a warning already telling people this is a bad idea.

Reply:
Hmm, agree about padding for sure. It still seems like a binary string of zeros should probably be False, but we're just defaulting to Python behavior, so they've probably thought about it more than I have in `bool(b'\x00')` is really evaluating whether there is any data in the byte string. I guess the way to actually check it's value would need to be at the bit level, like `bool(int.from_bytes(b'\x00', byteorder='big') & 0xFF)`.

All that to say, let's go with the Python behavior for evaluating the truthiness of binary strings. The XTCE spec even says on the subject of BooleanParameterType, "Binary encoding is typically a 'pass through' raw encoding form...".